test/e2e/notification_test.go

```diff
@@ -55,4 +57,22 @@ func (s *notificationTestSuite) TestNotificationCleanup() {
 		err = s.hostAwait.WaitUntilNotificationDeleted(mur.Name + "-provisioned")
 		require.NoError(t, err)
 	})
+
+	s.T().Run("deactivation notification created and deleted", func(t *testing.T) {
```

Comment:
I wonder if we actually would want to move this tests to our provisioning and deactivation flows.
Provisioning notification test IMO should be part of https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/user_assertions.go#L27:6
And deactivation notification tests should be part of https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/user_assertions.go#L27:6
Just to be clear. I'm not talking about adding a new test method to the existing suites. I'm talking about adding the notification status and resource checks in the existing corresponding tests. So it's tested as part of the existing flows instead of creating a new flow dedicated to the notification tests.

Reply:
So I couldn't move the provisioning notification test into `VerifyResourcesProvisionedForSignup` because we do multiple creations:

eg.
```
	johnsmithName := "johnsmith"
	johnSignup := CreateAndApproveSignup(t, hostAwait, johnsmithName)
	extrajohnName := "extrajohn"
	johnExtraSignup := CreateAndApproveSignup(t, hostAwait, extrajohnName)

	VerifyResourcesProvisionedForSignup(t, hostAwait, memberAwait, johnSignup, "basic")
	VerifyResourcesProvisionedForSignup(t, hostAwait, memberAwait, johnExtraSignup, "basic")
```

So the first call to `VerifyResourcesProvisionedForSignup` will work but the second call will fail because this function will return an error
 `notification, err := hostAwait.WaitForNotification(mur.Name+"-provisioned", wait.UntilNotificationHasConditions(Sent()))` 

This is because the second creation is already processed (notification created and deleted) by the time it's called so I simply moved it into `CreateAndApproveSignup` instead so that we're always verifying it whenever we're using that function to create the usersignup so it is part of the existing flows like we wanted.